tests/drivers/gpio/gpio_get_direction/src/main.c

```diff
@@ -6,108 +6,117 @@
+ZTEST_F(gpio_get_direction, test_disconnect)
+{
+	int rv;
+
+	fixture->flags = GPIO_DISCONNECTED;
+	common(fixture);
+
+	rv = gpio_pin_is_input(fixture->port, fixture->pin);
 	if (rv == -ENOSYS) {
+		/* gpio_pin_direction() is not supported in the driver */
 		ztest_test_skip();
 	}
 
 	zassert_equal(false, rv, "gpio_pin_is_input() failed: %d", rv);
 
-	rv = gpio_pin_is_output(port, pin);
+	rv = gpio_pin_is_output(fixture->port, fixture->pin);
 	zassert_equal(false, rv, "gpio_pin_is_output() failed: %d", rv);
 }
 
-static void test_input(void)
+ZTEST_F(gpio_get_direction, test_input)
 {
 	int rv;
-	const struct device *port;
-	gpio_pin_t pin = TEST_PIN;
-	gpio_flags_t flags = GPIO_INPUT;
+	fixture->flags = GPIO_INPUT;
 
-	port = device_get_binding(TEST_DEV);
-	zassert_not_null(port, "device " TEST_DEV " not found");
+	common(fixture);
 
-	rv = gpio_pin_configure(port, pin, flags);
-	zassert_equal(0, rv, "gpio_pin_configure() failed: %d", rv);
-
-	rv = gpio_pin_is_input(port, pin);
+	rv = gpio_pin_is_input(fixture->port, fixture->pin);
 	if (rv == -ENOSYS) {
+		/* gpio_pin_direction() is not supported in the driver */
 		ztest_test_skip();
 	}
 
 	zassert_equal(true, rv, "gpio_pin_is_input() failed: %d", rv);
 
-	rv = gpio_pin_is_output(port, pin);
+	rv = gpio_pin_is_output(fixture->port, fixture->pin);
 	zassert_equal(false, rv, "gpio_pin_is_output() failed: %d", rv);
 }
 
-static void test_output(void)
+ZTEST_F(gpio_get_direction, test_output)
 {
 	int rv;
-	const struct device *port;
-	gpio_pin_t pin = TEST_PIN;
-	gpio_flags_t flags = GPIO_OUTPUT | GPIO_OUTPUT_INIT_LOW;
+	fixture->flags = GPIO_OUTPUT | GPIO_OUTPUT_INIT_LOW;
 
-	port = device_get_binding(TEST_DEV);
-	zassert_not_null(port, "device " TEST_DEV " not found");
+	common(fixture);
 
-	rv = gpio_pin_configure(port, pin, flags);
-	zassert_equal(0, rv, "gpio_pin_configure() failed: %d", rv);
-
-	rv = gpio_pin_is_input(port, pin);
+	rv = gpio_pin_is_input(fixture->port, fixture->pin);
 	if (rv == -ENOSYS) {
+		/* gpio_pin_direction() is not supported in the driver */
 		ztest_test_skip();
 	}
 
 	zassert_equal(false, rv, "gpio_pin_is_input() failed: %d", rv);
 
-	rv = gpio_pin_is_output(port, pin);
+	rv = gpio_pin_is_output(fixture->port, fixture->pin);
 	zassert_equal(true, rv, "gpio_pin_is_output() failed: %d", rv);
 }
 
-static void test_input_output(void)
+ZTEST_F(gpio_get_direction, test_input_output)
 {
 	int rv;
-	const struct device *port;
-	gpio_pin_t pin = TEST_PIN;
-	gpio_flags_t flags = GPIO_INPUT | GPIO_OUTPUT | GPIO_OUTPUT_INIT_LOW;
+	fixture->flags = GPIO_INPUT | GPIO_OUTPUT | GPIO_OUTPUT_INIT_LOW;
 
-	port = device_get_binding(TEST_DEV);
-	zassert_not_null(port, "device " TEST_DEV " not found");
-
-	rv = gpio_pin_configure(port, pin, flags);
-	zassert_equal(0, rv, "gpio_pin_configure() failed: %d", rv);
+	common(fixture);
 
-	rv = gpio_pin_is_input(port, pin);
-	if (rv == -ENOSYS) {
+	rv = gpio_pin_is_input(fixture->port, fixture->pin);
+	if (rv == -ENOTSUP) {
```

Comment:
```suggestion
	if (rv == -ENOSYS) {
```

It looks like the description of `gpio_pin_is_input` needs to be corrected. It says: https://github.com/zephyrproject-rtos/zephyr/blob/3efce4c93938ebdd7d5d9b3ffdcd2b096490cea5/include/zephyr/drivers/gpio.h#L800-L803 while in `z_impl_gpio_port_get_direction` we've got: https://github.com/zephyrproject-rtos/zephyr/blob/3efce4c93938ebdd7d5d9b3ffdcd2b096490cea5/include/zephyr/drivers/gpio.h#L784-L785
Same story with `gpio_pin_is_output`.

Reply:
Good catch - I'll fix that in my current PR.